### PR TITLE
Blog App: validations and model specs

### PR DESCRIPTION
**Pull request by Lawmsangi**
### In this pull request, I implemented the following: 

**Add the following validations:**

- For the User model:
- Name must not be blank.
- PostsCounter must be an integer greater than or equal to zero.

**For the Post model:**
- Title must not be blank.
- Title must not exceed 250 characters.
- CommentsCounter must be an integer greater than or equal to zero.
- LikesCounter must be an integer greater than or equal to zero.

**Add unit specs for all of your models' methods and validations.**

### CODE REVIEW

File: spec/models/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  it 'LikesCounter must be an integer greater than or equal to zero' do
+    post.likes_counter = -1
+    expect(post).to_not be_valid
+  end
+
+  it 'CommentsCounter must be an integer greater than or equal to zero' do
+    subject.comments_counter = -1
+    expect(subject).to_not be_valid
+  end
```
Comment (mckent05) [inline spec/models/post_spec.rb:31]: - You have done well so far but according to the project requirements, you are required to test for all custom methods in your model. Please write tests for the `recent_comments` method in your post model. Please fix this 👍 👍

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  it 'is not valid without a name' do
+    subject.name = nil
+    expect(subject).to_not be_valid
+  end
+
+  it 'PostsCounter must be an integer greater than or equal to zero' do
+    subject.posts_counter = -1
+    expect(subject).to_not be_valid
+  end
+end
```
Comment (mckent05) [inline spec/models/user_spec.rb:21]: - You have done well so far but according to the project requirements, you are required to test for all custom methods in your model. Please write tests for the `recent_posts` method in your user model. Please fix this 👍 👍

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @Lawmsangi, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job adding validation and validation spec for your models 👍 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by codepantha — APPROVED** [review]
# STATUS: APPROVED 🟢 ✔️ 

## Hi @Lawmsangi 👋, 

Great job implementing the required changes 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights 🎯 
✔️ Descriptive PR
✔️ Added validations and model specs 📌 
✔️ No linter errors 🟢 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**